internal/auth/issuer/pam/pam_issuer.go

```diff
@@ -393,25 +393,46 @@ func (s *PAMOIDCProvider) handleAuthorizationCodeGrant(ctx context.Context, req
 	return tokenResponse, nil
 }
 
+// maskSensitiveValue truncates a sensitive value for logging/error messages
+// Shows first 8 characters and last 4 characters, with "..." in between
+func maskSensitiveValue(value string) string {
+	if value == "" {
+		return "<empty>"
+	}
+	if len(value) <= 12 {
+		return "***" // Don't reveal short values at all
+	}
+	return value[:8] + "..." + value[len(value)-4:]
+}
+
 // Authorize handles the authorization endpoint for authorization code flow
 func (s *PAMOIDCProvider) Authorize(ctx context.Context, req *pamapi.AuthAuthorizeParams) (*AuthorizeResponse, error) {
+	stateValue := lo.FromPtrOr(req.State, "")
+	maskedState := maskSensitiveValue(stateValue)
 	s.log.Debugf("Authorize: starting authorization request - ClientId=%s, RedirectUri=%s, ResponseType=%s, Scope=%s, State=%s",
-		req.ClientId, req.RedirectUri, req.ResponseType, lo.FromPtrOr(req.Scope, ""), lo.FromPtrOr(req.State, ""))
+		req.ClientId, req.RedirectUri, req.ResponseType, lo.FromPtrOr(req.Scope, ""), maskedState)
 
 	// Validate required fields
-	if req.ClientId == "" || req.RedirectUri == "" {
-		s.log.Warnf("Authorize: missing required fields - ClientId=%s, RedirectUri=%s", req.ClientId, req.RedirectUri)
-		return nil, errors.New(ErrorInvalidRequest)
+	var missingFields []string
+	if req.ClientId == "" {
+		missingFields = append(missingFields, "client_id")
+	}
+	if req.RedirectUri == "" {
+		missingFields = append(missingFields, "redirect_uri")
+	}
+	if len(missingFields) > 0 {
+		s.log.Warnf("Authorize: missing required fields - %v", missingFields)
+		return nil, fmt.Errorf("%s: missing required parameter(s): %s", ErrorInvalidRequest, strings.Join(missingFields, ", "))
 	}
 
 	// Validate client ID
 	if s.config == nil {
 		s.log.Errorf("Authorize: config is nil")
-		return nil, errors.New(ErrorInvalidClient)
+		return nil, fmt.Errorf("%s: PAM issuer configuration not initialized", ErrorInvalidClient)
```

Comment:
ErrorInvalidClient I would expect to be an error type. I know you didnt add that but it feels awkward. 

```go 
const ErrInvalidClient =  errors.New("invalid client")

return fmt.Errorf("%w: PAM issuer configuration not initialized", ErrInvalidClient)
```

this allows golang level unwrapping in tests etc.
```go
if errors.Is(err, ErrInvalidClient) { // hangle error }